hudi-utilities/src/main/java/org/apache/hudi/utilities/keygen/TimestampBasedKeyGenerator.java

```diff
@@ -51,6 +51,8 @@
 
   private final String outputDateFormat;
 
+  private TimeZone timeZone;
```

Comment:
any reason why this is not `final` too?

Reply:
Because TimeZone is not necessarily initialized when the class is initialized.

Reply:
fix NPE, thank's viewed